pages/workoutlist.js

```diff
@@ -40,36 +51,67 @@ export default function WorkoutsList() {
       <StyledHeadline>Choose Your Workout</StyledHeadline>
 
       <WorkoutList>
-        {preparedWorkouts.map((workout) => {
-          const isDetailsVisible = showDetails[workout.id] || false;
-          return (
-            <WorkoutCard key={workout.id}>
-              <h2>{workout.name}</h2>
-              <SpotlightHeading>Muscles In The Spotlight:</SpotlightHeading>
-              <MuscleGroupList>
-                {workout.muscleGroups.map((muscle) => (
-                  <MuscleBadge key={muscle}>{muscle}</MuscleBadge>
-                ))}
-              </MuscleGroupList>
-              <ToggleButton onClick={() => toggleDetails(workout.id)}>
-                {isDetailsVisible ? "Show Less" : "Show More"}
-              </ToggleButton>
-              {isDetailsVisible && (
-                <ExerciseList>
-                  {workout.exercises.map((exercise) => {
-                    return (
-                      <ExerciseItem key={exercise.exerciseId}>
-                        <ExerciseName>{exercise.name}</ExerciseName>
-                        <ExerciseDetails>Sets: {exercise.sets}</ExerciseDetails>
-                        <ExerciseDetails>Reps: {exercise.reps}</ExerciseDetails>
-                      </ExerciseItem>
-                    );
-                  })}
-                </ExerciseList>
-              )}
-            </WorkoutCard>
-          );
-        })}
+        {!preparedWorkouts.length ? (
+          <NoWorkoutsMessage>
+            No workouts yet! <br />
+            <br />
+            Create a workout and reach your goals!
+          </NoWorkoutsMessage>
+        ) : (
+          preparedWorkouts.map((workout) => {
+            const isDetailsVisible = showDetails[workout.id] || false;
+            return (
+              <WorkoutCard key={workout.id}>
+                {deleteMode === workout.id ? (
+                  <ModalOverlay>
+                    <ModalContent>
+                      <p>Are you sure you want to delete?</p>
+                      <YesButton onClick={() => handleDelete(workout.id)}>
+                        YES
+                      </YesButton>
+                      <ModalButton onClick={() => toggleDeleteMode(null)}>
```

Comment:
See my comment above regarding your `toggleDeleteMode` function.

```suggestion
                      <ModalButton onClick={() => setWorkoutToDelete(workout.id)}>
```